controllers/tasksRequests.js

```diff
@@ -0,0 +1,95 @@
+const { INTERNAL_SERVER_ERROR, SOMETHING_WENT_WRONG } = require("../constants/errorMessages");
+const taskRequestsModel = require("../models/taskRequests");
+
+const fetchTaskRequests = async (_, res) => {
+  try {
+    const taskRequests = await taskRequestsModel.fetchTaskRequests();
+
+    return res.status(200).json({
+      message: "Task requests returned successfully",
+      taskRequests,
+    });
+  } catch (err) {
+    logger.error("Error while fetching task requests", err);
+    return res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
+};
+
+const addOrUpdate = async (req, res) => {
+  try {
+    const { taskId, userId } = req.body;
+
+    if (!taskId) {
+      return res.boom.badRequest("taskId not provided");
+    }
+    if (!userId) {
+      return res.boom.badRequest("userId not provided");
+    }
+
+    const response = await taskRequestsModel.addOrUpdate(taskId, userId);
+    if (response.userDoesNotExists) {
+      return res.boom.conflict("User does not exist");
+    }
+    if (response.requestorExists) {
+      return res.boom.conflict("User is already requesting for the task");
+    }
+    if (response.taskRequestDoesNotExist) {
+      return res.boom.conflict("Task request already exist");
+    }
+
+    if (response.isUpdate) {
+      return res.status(200).json({
+        message: "Task request successfully updated",
+        requestors: response.requestors,
+      });
+    }
+
+    return res.status(201).json({
```

Comment:
Either use status or boom here to maintain consistency.

Reply:
same as above